src/checking.rs

```diff
@@ -152,14 +155,44 @@ async fn check_impl(
         return None;
     }
 
-    let response = match if check_level == CheckLevel::ForLinks {
-        CLIENT.get(website.to_string()).send().await
-    } else {
-        CLIENT.head(website.to_string()).send().await
-    } {
-        Ok(response) => response,
+    let mut response;
+
+    let mut retry_limit = 5;
+
+    loop {
+        response = if check_level == CheckLevel::ForLinks {
+            CLIENT.get(website.to_string()).send().await
+        } else {
+            CLIENT.head(website.to_string()).send().await
+        };
+
+        if retry_limit == 0 {
+            break;
+        }
+
+        match &response {
+            Ok(_) => break,
+            Err(err) => {
+                warn!(
+                    "Could not request website {website} with error {err}. Retrying after five seconds."
+                );
+            }
+        }
+
+        retry_limit -= 1;
+
+        // Don't stall our testcases
+        #[cfg(not(test))]
```

Comment:
Maybe it would be better to use Tokio's time mocking features? See https://tokio.rs/tokio/topics/testing#pausing-and-resuming-time-in-tests.